src/elements.hpp

```diff
@@ -59,6 +59,7 @@ class G1Element {
     GTElement Pair(const G2Element &b) const;
     uint32_t GetFingerprint() const;
     std::vector<uint8_t> Serialize() const;
+    G1Element Copy();
```

Comment:
in C++, this functionality is implemented by copy constructors. I can't see a good reason to have this be a member function instead. But, if there is a good reason, it should *at least* be a `const` member function